### PR TITLE
Add new sharepoint sites service

### PR DESCRIPTION
**Pull request by jnormington**
Adds a new siteService for fetching Sharepoint sites, which implement the SiteService interface. 
Which will make it simpler to build mocks for testing in spreadsheets.

### CODE REVIEW

File: site_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+func TestSiteService_Search(t *testing.T) {
+	t.Run("successfully returns a collection of sites", func(t *testing.T) {
+		setup()
+		defer teardown()
+
+		srv := siteService{client: oneDrive}
+		mux.HandleFunc("/sites", func(w http.ResponseWriter, req *http.Request) {
+			assert.Equal(t, req.URL.Query().Get("search"), "the filter")
+			fileWrapperHandler("fixtures/site.search.valid.json", 200)(w, req)
+		})
```
Comment (BRMatt) [inline site_test.go:22]: Is `oneDrive` a shared instance of the client associated with this globally mutable http server?

**Review by BRMatt — APPROVED** [review]
Code seems ok, but the PR doesn't really have a lot of context for why this change is necessary, or how it will be used....

File: site_test.go (hunk 1 above)
Comment (jnormington) [reply site_test.go:22]: Yeah unfortunately it is in the onedrive_test.go file

**Comment by jnormington** [comment]
> Code seems ok, but the PR doesn't really have a lot of context for why this change is necessary, or how it will be used....

Its for the new sharepoint service within spreadsheets, and will be used to fetch the users sharepoint sites, which will be required to be selected first before any files within the sharepoint provider will be be returned.

**Pull request merged**